### ISSUE TITLE
[BUG] Ports duplicated when using extends

### ISSUE THREAD

**Issue by kierans** [docker/compose#11611]
### Description

When using `extends` on a service between two `docker-compose.yaml` files, the `ports` array is duplicated/merged resulting in an array where ports are listed twice. Therefore, when booting the container Docker complains about `bind: address already in use`

I’m using [Devcontainers][1] for this project, so I have the base project Compose config and a developer specific config

[1]: https://containers.dev/

### Steps To Reproduce

```yaml

# docker-compose.yaml

version: '3.8'

services:
  my-service:
    build:
      context: .
      target: service
    depends_on:
      - db
    ports:
      - "8080:8080"
```

```yaml

# .devcontainer/docker-compose-dev.yaml

version: '3.8'

services:
  my-service:
    extends:
      file: docker-compose.yaml
      service: my-service
    build:
      target: dev
    volumes:
      - .:/app
```

Looking at the config, the port mapping of `8080` is duplicated

```shell
$ cd .devcontainer
$ docker compose -f ../docker-compose.yaml -f docker-compose-dev.yaml config | less
```

```yaml
ports:
  - mode: ingress
    target: 8080
    published: "8080"
    protocol: tcp
  - mode: ingress
    target: 8080
    published: "8080"
    protocol: tcp
```

When the container launches, I get the error

```
...
Error response from daemon: Ports are not available: exposing port TCP 0.0.0.0:8080 -> 0.0.0.0:0: listen tcp 0.0.0.0:8080: bind: address already in use
```

### Compose Version

```Text
Docker Desktop 4.27.2
Docker Compose version v2.24.5-desktop.1
```

### Anything else?

This does work properly in `Docker Desktop 4.26` (`Docker Compose version v2.23.3-desktop.2`)

**Comment by ndeloof** [comment]
This has been fixed already, see https://github.com/compose-spec/compose-go/pull/529

**Comment by kierans** [comment]
@ndeloof I’m not sure this is fixed, perhaps there is a regression. Or I’m just missing something.

https://github.com/compose-spec/compose-go/pull/529 was released in `compose-spec#v2.0.0-rc.1` which was integrated into `compose#v2.24.1` (https://github.com/docker/compose/commit/c28bf5227aafe47db68423437cc1371786d3c577). However, I was experiencing the issue in Docker Desktop `4.27.2` which is bundled with Docker Compose `v2.24.5-desktop.1` (which uses compose-spec `v2.0.0-rc.3` in `go.mod`)
